src/main/java/ch/wisv/areafiftylan/service/UserServiceImpl.java

```diff
@@ -68,7 +69,9 @@ public User getUserById(long id) {
     }
 
     @Override
-    public User create(UserDTO userDTO, HttpServletRequest request) {
+    public User create(UserDTO userDTO, HttpServletRequest request) throws DataIntegrityViolationException {
+        checkIfFieldsInUse(userDTO);
```

Comment:
A check should return a boolean value and not 'blindly' throw an Exception in my eyes. The responsibility is a check, not to handle that check. If that would be the case the function could be renamed for better code readability to "checkIfFieldInUseAndHandleResult" or something which is better. I would still prefer to return a boolean or maybe even an enum that returns what is in use: "None", "Username" or "Email".